src/utils.ts

```diff
@@ -67,11 +69,18 @@ export function getWalletWithProvider({
   /* c8 ignore start */
   if (!provider) {
     // This will be significantly rate limited, but we only need to run it once
-    provider = getDefaultProvider(network);
+    provider = getDefaultProvider({ ...network, name: network.chainName });
   }
+
   if (!provider) {
     throw new Error("unable to create ETH API provider");
   }
+  (async () => {
```

Comment:
This isn't going to be safe. We'll have to update this method to be async. That kinda sucks, but async is infectious, so we'll need to update all the calls to it as well. Bummer.

Reply:
> Just updating my review based on the async issue in the utils.

Updated. The Connections (created during setup) class is now only usable once a ready check has completed, during which it checks chainId compatibility.